### ISSUE TITLE
Use the function "render_to_image" to traverse to get the picture, which will cause memory overflow?

### ISSUE THREAD

**Issue by lfhgljj** [intel-isl/Open3D#2966]
My purpose is to match the target object in the two-dimensional image with the planar projection of the target object in the three-dimensional space. I have successfully implemented it through pnp. Now, I want to try other methods.

I plan to use two different loss functions and compare the results.
（1）loss = 1 - iou
（2）loss =  np.sum((image_mask - model_projection_mask)**2)
 But the above method requires curved surface projection。

when I use the 'vis.capture_screen_float_buffer()', I can get the correct result .But to use this function, I need to create a window.
Now I find "render.render_to_image()",But using it to traverse will cause memory overflow and kill the program.

I'm pretty sure that I haven't changed other places, just replaced "vis.capture_screen_float_buffer" with "render.render_to_image", but I didn't find the reason, I want to ask for your help, thank you.

My guess is that the window opened by ‘open3d.visualization.rendering.OffscreenRenderer’ is not destroyed, but I don’t see the function for destruction.

`
This is the changed code:

      def draw_model(mesh, shape):
          vis = Visualizer()
          vis.create_window(width=shape[1], height=shape[0])
          vis.get_render_option().background_color = np.asarray([0, 0, 0])
          vis.get_render_option().mesh_show_back_face = True
    
          view_ctl = vis.get_view_control()
          vis.add_geometry(mesh)
          cam = view_ctl.convert_to_pinhole_camera_parameters()
          cam.extrinsic = pose
          view_ctl.convert_from_pinhole_camera_parameters(cam)
      
          # vis.run()
          img = vis.capture_screen_float_buffer()
          vis.destroy_window()
      
          # cv2.imshow("foot_model", np.array(img))
          # cv2.waitKey()
          return np.array(img)

      def draw_model(mesh, shape, visible=False):
          render = rendering.OffscreenRenderer(shape[1], shape[0], headless=True)
          mtl = rendering.Material()
          mtl.base_color = [1, 1, 1, 1]
          mtl.shader = "defaultLit"

          render.scene.set_background([0, 0, 0, 0])
          render.scene.add_geometry("foot_model", mesh, mtl)
          render.scene.set_lighting(render.scene.LightingProfile.NO_SHADOWS, (0, 0, 0))
      
          render.scene.camera.look_at([0, 0, 0], [0, 10, 0], [0, 0, 1])
          img_o3d = render.render_to_image()
          img = np.array(img_o3d)
      
          if vis:
              cv2.imshow("foot_model", img)
              cv2.waitKey()
      
          return img
`

**Comment by prewettg** [comment]
The following works for me:
```
import numpy as np
import open3d as o3d
import cv2

torus = o3d.geometry.TriangleMesh.create_torus()
torus.compute_vertex_normals()
torus.paint_uniform_color([0.1, 0.7, 0.4])

render = o3d.visualization.rendering.OffscreenRenderer(640, 480, headless=False)
mtl = o3d.visualization.rendering.Material()
mtl.base_color = [1, 1, 1, 1]
mtl.shader = "defaultLit"

render.scene.set_background([0, 0, 0, 0])
render.scene.add_geometry("model", torus, mtl)
render.scene.set_lighting(render.scene.LightingProfile.NO_SHADOWS, (0, 0, 0))
  
render.scene.camera.look_at([0, 0, 0], [0, 10, 0], [0, 0, 1])
img_o3d = render.render_to_image()
img = np.array(img_o3d)
  
cv2.imshow("foot_model", img)
cv2.waitKey()
```

`headless=True` is only for true headless machines that have no window system at all (and is limited to Linux, although partially that is because Windows and macOS always have a window system). Since you are using cv2.imshow() you must have a window system, so you should be using `headless=False`.
